oak-core/src/main/java/org/apache/jackrabbit/oak/query/QueryEngineSettings.java

```diff
@@ -80,6 +82,8 @@
     private StrictPathRestriction strictPathRestriction = StrictPathRestriction.DISABLE;
 
     private final QueryStatsMBeanImpl queryStats = new QueryStatsMBeanImpl(this);
+    
+    private String[] classNamesIgnoredInCallTrace = new String[0];
```

Comment:
sounds like a reasonable default..... and when reading it i noticed about the missing @nonull annotations.... because the default might get replaced by 'null' upon set if you don't specify it to be not-null.....